[AN-Issue-1485] Enabled means to configure automation registry conditionally or manually

aregng opened the pull request:
- Added a new `AutomationRegistryConfig` configuration type  enclosing
  registry configuration parameters

- Updated `GenesisConfigurations` with optional automation-registry-configuration paramter

- Introduced  function `genesis::initialize_supra_native_automation` to
  initialize auomation registry externally

- Initialized  `automation_registry` during genesis transcation
  generation conditionaly based on corresponding configuration paramters
  availability

- Introduced  feature flag which will allow conditionally fully enable
  supra native automation usage
  
- Guarded automation registry functions execution 
    - `automation_registry::on_new_epoch` will do noop if registry is not properly initialized
    - `automation_registry::register` will fail task registration if `SUPRA_NATIVE_AUTOMATION` is not enabled

Commit:
[AN-Issue-1419] Updated Automation Registration Parameters

- Added 2 new properties
  - automation_fee_cap_for_epoch to allow user decide on threshold for the automation fee paied for epoch
  - aux_data - means to allow future minor registration parameters integration requiring only framework/state upgrade
- Refactor RegistrationParams to allow easy extend with new/other set of parameter lists in future

Commit:
Addressed reivew comments

Commit:
[AN-Issue-1485] Enabled means to configure automation registry conditionally or manually

- Added a new `AutomationRegistryConfig` configuration type  enclosing
  registry configuration parameters

- Updated `GenesisConfigurations` with optional automation-registry-configuration paramter

- Introduced  function `genesis::initialize_supra_native_automation` to
  initialize auomation registry externally

- Initialized  `automation_registry` during genesis transcation
  generation conditionaly based on corresponding configuration paramters
  availability

- Introduced  feature flag which will allow conditionally fully enable
  supra native automation usage

Commit:
Fixed config existence check in \'automation-registry::update_epoch_interval_in_registry`

Commit:
Guarded automation registry functions execution

- `automation_registry::on_new_epoch` will do noop if registry is not properly initialized
- `automation_registry::register` will fail task registration if `SUPRA_NATIVE_AUTOMATION` is not enabled

Commit:
Fixed the typo

aregng committed:
Merge branch 'feature/automation-networks' into task/issue-1485-feature-flag

nizam-supraoracles commented on aptos-move/framework/supra-framework/sources/automation_registry.move:
```diff
@@ -194,6 +216,30 @@ module supra_framework::automation_registry {
+    }
+
+    /// Initialization of Automation Registry with configuration parameters is expected metrics.
+    public(friend) fun initialize_internal(
```
In my opinion, we shouldn't make this function `public(friend)` and should keep it private.
I see that you are utilizing this function in the genesis module, but instead, we can modify the `initialize` function and make it `public(friend)` and use in genesis module.

aregng replied:
Fixed

nizam-supraoracles reviewed (approved):
Only one remark—if it's feasible, we can make the change. Everything else looks good from my end.

Commit:
Cosmetic changes

Commit:
Added view functions
 - Added view functions to query registered tasks count and ids
 - Added view functions to query registry initialization status

Commit:
Fixed test failures and removed redundant code

Commit:
Updated SUPRA_NATIVE_AUTOMATION flag value to 88

Commit:
Updated features readme

Commit:
Cosmentic changes

- make the genesis::initialize_automation_registry public to allow calls form script
- fixed couple of typos in the code and warnings
